### PR TITLE
Test on php 7.4, improve static analysis

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

**Comment by chives** [comment]
@fsi-open

File: DependencyInjection/Compiler/FirewallMapCompilerPass.php [hunk 1]
```diff
@@ -21,8 +21,10 @@ class FirewallMapCompilerPass implements CompilerPassInterface
 
     public function process(ContainerBuilder $container): void
     {
-        if (false === $container->hasAlias(self::FIREWALL_MAPPER_SERVICE) &&
-            false === $container->hasDefinition(self::FIREWALL_MAPPER_SERVICE)) {
+        if (
+            false === $container->hasAlias(self::FIREWALL_MAPPER_SERVICE) &&
```
Comment (szymach) [inline DependencyInjection/Compiler/FirewallMapCompilerPass.php:25]: `&&` should be on the next line.

**Review by szymach — APPROVED** [review]

**Comment by chives** [comment]
@fsi-open

**Comment by chives** [comment]
@fsi-open now it's really ready

File: Controller/Activation/ActivationController.php [hunk 2]
```diff
@@ -110,21 +111,18 @@ public function activateAction(string $token): Response
             return new RedirectResponse(
                 $this->router->generate('fsi_admin_activation_change_password', ['token' => $token])
             );
-        } else {
-            $this->eventDispatcher->dispatch(
-                AdminSecurityEvents::ACTIVATION,
-                new ActivationEvent($user)
-            );
-
-            return $this->addFlashAndRedirect('success', 'admin.activation.message.success');
```
Comment (szymach) [inline Controller/Activation/ActivationController.php:119]: Well if you really want to refactor this, you could leave the `if/else` but assign the response to a variable, so there's only one `return` statement.

File: Doctrine/Repository/UserRepository.php [hunk 3]
```diff
@@ -13,15 +13,34 @@
...
     {
-        return $this->findOneBy(['activationToken.token' => $activationToken]);
+        $user = $this->findOneBy(['activationToken.token' => $activationToken]);
+        if (null === $user) {
+            return null;
+        }
+        if (false === $user instanceof ActivableInterface || false === $user instanceof ChangeablePasswordInterface) {
+            throw new RuntimeException(
+                sprintf(
+                    "Expected to get instance of %s and %s but got instance of %s",
```
Comment (szymach) [inline Doctrine/Repository/UserRepository.php:35]: `Class "%s" does not implement both "%s" and "%s"`

**Comment by chives** [comment]
@fsi-open

**Pull request merged**